src/tests/ftest/io/NvmeIO.py

```diff
@@ -0,0 +1,203 @@
+        self.d_log = DaosLog(self.context)
+        self.hostlist = self.params.get("servers", '/run/hosts/*')
+        self.hostfile = WriteHostFile.WriteHostFile(self.hostlist,
+                                                    self.workdir)
+        #Start Server
+        ServerUtils.runServer(self.hostfile, self.server_group, self.basepath)
+
+    def tearDown(self):
+        try:
+            if self.pool_connect:
+                self.pool.disconnect()
+                self.pool.destroy(1)
+        finally:
+            ServerUtils.stopServer(hosts=self.hostlist)
+
+    def verify_pool_size(self, original_pool_info, ior_args):
+        """
+        Function is to validate the pool size
+        original_pool_info: Pool info prior to IOR
+        ior_args: IOR args to calculate the file size
+        """
+        #Get the current pool size for comparison
+        current_pool_info = self.pool.pool_query()
+        #if Transfer size is < 4K, Pool size will verified against NVMe, else
+        #it will be checked against SCM
+        if ior_args['stripe_size'] >= 4096:
+            print("Size is > 4K,Size verification will be done with NVMe size")
+            storage_index = 1
+        else:
+            print("Size is < 4K,Size verification will be done with SCM size")
+            storage_index = 0
+
+        free_pool_size = (
+            original_pool_info.pi_space.ps_space.s_free[storage_index]
+            - current_pool_info.pi_space.ps_space.s_free[storage_index])
+        expected_pool_size = ior_args['slots'] * ior_args['block_size']
+        if free_pool_size < expected_pool_size:
+            raise DaosTestError(
+                'Pool Free Size did not match Actual = {} Expected = {}'
+                .format(free_pool_size, expected_pool_size))
+
+    @avocado.fail_on(DaosApiError)
+    def test_nvme_io(self):
+        """
+        Test ID: DAOS-2082
+        Test Description: Test will run IOR with standard and non standard
+        sizes.IOR will be run for all Object type supported. Purpose is to
+        verify pool size (SCM and NVMe) for IOR file.
+        This test is running multiple IOR on same server start instance.
+        :avocado: tags=nvme,nvme_io,large
+        """
+        total_ior_threads = 1
+        ior_args = {}
+
+        hostlist_clients = self.params.get("clients", '/run/hosts/*')
+        tests = self.params.get("tests", '/run/ior/*')
+        object_type = self.params.get("object_type", '/run/ior/*')
+        #Loop for every IOR object type
+        for obj_type in object_type:
+            for ior_param in tests:
+                self.hostfile_clients = WriteHostFile.WriteHostFile(
+                    hostlist_clients,
+                    self.workdir,
+                    ior_param[4])
+                #There is an issue with NVMe if Transfer size>64M, Skipped this
+                #sizes for now
+                if ior_param[2] > 67108864:
```

Comment:
you could use cancel/skip built-in function

Reply:
Skip will skip the complete test, But in this case it's all single test so can not use skip/cancel.